Client/Application/AppDelegate.swift

```diff
@@ -54,6 +54,13 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Force a database upgrade by requesting a non-existent password
         profile.logins.getLoginsForProtectionSpace(NSURLProtectionSpace(host: "example.com", port: 0, `protocol`: nil, realm: nil, authenticationMethod: nil))
 
+        // check to see if we started cos someone tapped on a notification
+        if let localNotification = launchOptions?[UIApplicationLaunchOptionsLocalNotificationKey] as? UILocalNotification {
+            if let alertURL = localNotification.alertBody {
+                browserViewController.openURLInNewTab(NSURL(string: alertURL)!)
```

Comment:
Since the alertBody is coming from an unknown source (the network), I think we should check if the `NSURL` can actually be constructed. Just put it in the `if let` and not do anything if the incoming URL was invalid/malformed?